OIE by default google authenticator java (#3152)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/challengeintegrationsteps.md

### 1 - 4: Sign-in and Select Authenticator

The challenge flow follows the same first four steps as the [enrollment flow](#integrate-sdk-for-authenticator-enrollment):

* Build a sign-in page on the client
* Authenticate the user credentials
* Handle the response from the sign-in flow
* Display a list of possible authenticator factors

### 5: Check Authenticator Status

> When the user selects the Google Authenticator factor, call the `selectAuthenticator()` method, to tell Okta which factor is being used.

Author: ```suggestion
When the user selects Google Authenticator, update Okta with the chosen factor by calling the `selectAuthenticator()` method.
```

> When in Challenge Flow, a call is made to `idxClient.SelectChallengeAuthenticatorAsync`, using its `selectAuthenticatorOptions` parameter to pass in the Google Authenticator factor ID.

Reviewer: What is "Google Authenticator factor ID"? Is it the TOTP entered by the user? Or is it informing Okta of the selected authenticator, in which case, what's the differentiation with the sentence above (it can't be simply challenge flow because this is the challenge flow.) It also doesn't seem necesary to call out "when in challenge flow" because this is whole document is about the challenge flow.

```java
case AWAITING_AUTHENTICATOR_SELECTION:
    // get the possible list of factors and have the user select one
    // authenticationResponse.getAuthenticators()

    // continue with the selected factor
    authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
```

> > **NOTE:** If your application's sign-on policy only contains one factor, this state will be skipped.

Author: ```suggestion
> **NOTE:**  This state is skipped if your application's sign-on policy contains only one factor.
```

> ### 1 - 4: Sign-in and Select Authenticator

Reviewer 2: ```suggestion
### 1 - 4: Sign in and select authenticator
```

The challenge flow follows the same first four steps as the [enrollment flow](#integrate-sdk-for-authenticator-enrollment):

* Build a sign-in page on the client
* Authenticate the user credentials
* Handle the response from the sign-in flow
* Display a list of possible authenticator factors

> ### 5: Check Authenticator Status

Reviewer 2: ```suggestion
### 5: Check authenticator status
```

When the user selects the Google Authenticator factor, call `selectAuthenticator()`, to tell Okta which factor is being used.

> > **Note:** For Email or SMS factors, this sends a message to the user.

Reviewer 3: This note doesn't seem necessary as this isn't a general guide. We're only talking about the Google factor,, not email or SMS

> > **NOTE:** For Email or SMS factors this will send a message to the user.

Reviewer 2: ```suggestion
> **Note:** For Email or SMS factors, this sends a message to the user.
```

> When in Challenge Flow, a call is made to the `idxAuthenticationWrapper.selectAuthenticator` method, using the corresponding `Authenticator` object.

Author: ```suggestion
When in Challenge Flow, a call is made to the `idxAuthenticationWrapper.selectAuthenticator()` method, using the corresponding `Authenticator` object.
```

> When in Challenge Flow, a call is made to `idxClient.SelectChallengeAuthenticatorAsync`, using its `selectAuthenticatorOptions` parameter to pass in the Google Authenticator factor ID.

Reviewer 4: `verifyAuthenticator(..)` is intended here instead?

> When in Challenge Flow, a call is made to `idxClient.SelectChallengeAuthenticatorAsync`, using its `selectAuthenticatorOptions` parameter to pass in the Google Authenticator factor ID.

Author: Good catch, it looks like a bit of copypasta from the .net guide

```suggestion
When in Challenge Flow, a call is made to the `idxAuthenticationWrapper.selectAuthenticator` method, using the corresponding `Authenticator` object.
```

```java
case AWAITING_AUTHENTICATOR_SELECTION:
    // get the possible list of factors and have the user select one
    // authenticationResponse.getAuthenticators()

    // continue with the selected factor
    authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
```

> > **NOTE:** If your application's sign-on policy only contains one factor, this state is skipped.

Reviewer 2: ```suggestion
> **Note:** If your application's authentication policy contains only one factor, this state is skipped.
```

### 6: Get one-time password from Google Authenticator

> The user’s copy of Google Authenticator now displays the time-based one-time password (TOTP) for the newly added account, which they will enter into a challenge page.

Reviewer 2: ```suggestion
The user’s copy of Google Authenticator now displays the time-based one-time password (TOTP) for the newly added account that they enter into a challenge page.
```

> The user’s copy of Google Authenticator now displays the time-based one-time password for the newly added account which they will enter into a challenge page.

Reviewer 5: ```suggestion
The user’s copy of Google Authenticator now displays the time-based one-time password (TOTP) for the newly added account, which they will enter into a challenge page.
```

![A one-time password being shown in Google Authenticator](/img/authenticators/authenticators-google-one-time-password.png)

> After the user selects Google Authenticator and prompts the for a TOTP (the _code_), verify the code with Okta by calling the `verifyAuthenticator` method:

Reviewer 2: ```suggestion
After the user selects Google Authenticator and prompts the user for a TOTP (the _code_), verify the code with Okta by calling the `verifyAuthenticator` method:
```

> After the user selects Google Authenticator and prompts the for a time-based one-time password (the _code_), verify the code with Okta by calling the `verifyAuthenticator` method:

Reviewer 5: ```suggestion
After the user selects Google Authenticator and prompts the for a TOTP (the _code_), verify the code with Okta by calling the `verifyAuthenticator` method:
```

```java
case AWAITING_AUTHENTICATOR_VERIFICATION:

>     // confirm the TOTP code with Okta and to back into the state machine

Reviewer 2: ```suggestion
    // confirm the TOTP code with Okta and to go back into the state machine
```

    authenticationResponse = idxAuthenticationWrapper
        .verifyAuthenticator(proceedContext, new VerifyAuthenticatorOptions(code));
```

### 7: Handle the Success status.

> A user is only successfully authenticated once the `AuthentiationStatus` is `SUCCESS`. After a user is authenticated, If your application needs to make other HTTP requests on behalf of the user, you can access their ID and access tokens by calling the `getTokenResponse()` object.

Reviewer 2: ```suggestion
A user is only successfully authenticated after the `AuthenticationStatus` is `SUCCESS`. After a user is authenticated, if your application needs to make other HTTP requests on behalf of the user, you can access their ID and access tokens by calling the `getTokenResponse()` object.
```

> A user is only successfully authenticated once the `AuthentiationStatus` is `SUCCESS`. After a user is authenticated, If your application needs to make other HTTP requests on behalf of the user, you can access their ID and access tokens, by calling the `getTokenResponse()` object.

Reviewer 5: ```suggestion
A user is only successfully authenticated once the `AuthentiationStatus` is `SUCCESS`. After a user is authenticated, If your application needs to make other HTTP requests on behalf of the user, you can access their ID and access tokens by calling the `getTokenResponse()` object.
```
